polliwog/polyline/_polyline_object.py

```diff
@@ -587,24 +587,31 @@ def nearest(
             indices_of_nearest_segments.reshape(num_points, 1, 1),
             axis=1,
         ).reshape(num_points, 3)
-        t_values_of_closest_points = np.take_along_axis(
-            t_values,
-            indices_of_nearest_segments.reshape(num_points, 1),
-            axis=1,
-        )
 
         if ret_segment_indices or ret_distances:
             result = [transform_result(closest_points_of_polyline)]
             if ret_segment_indices:
                 result.append(transform_result(indices_of_nearest_segments))
             if ret_distances:
                 result.append(
-                    transform_result(distance_to_closest_points_of_segments)[
-                        indices_of_nearest_segments
-                    ]
+                    transform_result(
+                        np.take_along_axis(
+                            distance_to_closest_points_of_segments,
+                            indices_of_nearest_segments.reshape(num_points, 1),
+                            axis=1,
+                        ).flatten()
+                    )
                 )
             if ret_t_values:
-                result.append(transform_result(t_values_of_closest_points))
+                result.append(
+                    transform_result(
+                        np.take_along_axis(
+                            t_values,
+                            indices_of_nearest_segments.reshape(num_points, 1),
+                            axis=1,
+                        )
```

Comment:
It’s strange to me that one of these has flatten and the other doesn’t. Are the shapes changing with the call to flatten()?

Reply:
hmm... good point.  I'll look into this this afternoon.

Reply:
yeah, I had to add flatten to the t_values as well.  There wasn't coverage for that case... so I added that too.